src/shogun/regression/svr/SVRLight.h

```diff
@@ -186,6 +180,13 @@ class CSVRLight: public CSVMLight
 			int32_t *docs,float64_t *aicache,
 			float64_t* maxdiff);
 
+		/** @return whether this svm solves a regression problem or classification */
+		virtual bool is_regression() const 
```

Comment:
supports()

Reply:
this should go. I don't think supports is needed now, only "supports_interleaving" is left, and the rest comes from EProblemType

Reply:
`supports_interleaving` should go and add the described supports() function...

Reply:
there we can add what sort of input type it supports etc.... way more extensible than having a simple function that returns a boolean for a certain small thing

Reply:
and that interface should actually go into CMachine

Reply:
it is what EProblemType is for I think

Reply:
The CMachine is general purpose, but supports_interleaving only concerns MKL, is this ok? Also problem type and label type currently have their own enums (ELabelType and EMachineType), and they correspond. Sorry just want to make sure

Reply:
you can encode in one bit whether a machine is interleaving or not...